src/redux/cartRedux.tsx

```diff
@@ -1,43 +1,69 @@
+import { Store } from "./initialState";
+import { CartItem } from "../types/CartItem";
+
 //selectors
-export const getCartProducts = (state) => state.cart;
+export const getCartProducts = (state: Store) => state.cart;
 
-// action name creator
-const createActionName = (actionName) => `app/accessories/${actionName}`;
+// action creators
+export const addToCart = (payload: CartItem): AddItemAction => ({ type: "ADD_TO_CART", payload });
+export const addAmount = (id: string, amount: number): AddItemAmountAction => ({ type: "ADD_AMOUNT", id, amount });
+export const removeAmount = (payload: RemoveAmountAction) => ({ type: "REMOVE_AMOUNT", payload });
+export const removeItem = (id: string): RemoveItemAction => ({ type: "REMOVE_ITEM", id });
+export const clearCart = (): ClearCartAction => ({ type: "CLEAR_CART" });
 
-// action types
-const ADD_TO_CART = createActionName("ADD_TO_CART");
-const ADD_AMOUNT = createActionName("ADD_AMOUNT");
-const REMOVE_AMOUNT = createActionName("REMOVE_AMOUNT");
-const REMOVE_ITEM = createActionName("REMOVE_ITEM");
-const CLEAR_CART = createActionName("CLEAR_CART");
 
-// action creators
-export const addToCart = (payload) => ({ type: ADD_TO_CART, payload });
-export const addAmount = (payload) => ({ type: ADD_AMOUNT, payload });
-export const removeAmount = (payload) => ({ type: REMOVE_AMOUNT, payload });
-export const removeItem = (payload) => ({ type: REMOVE_ITEM, payload });
-export const clearCart = (payload) => ({ type: CLEAR_CART, payload });
+//action types
+interface AddItemAction {
+  type: "ADD_TO_CART"
+  payload: CartItem
+}
+
+interface AddItemAmountAction {
+  type: "ADD_AMOUNT"
+  id: string
+  amount: number
+}
+
+interface RemoveItemAction {
+  type: "REMOVE_ITEM"
+  id: string
+}
+
+interface ClearCartAction {
+  type: "CLEAR_CART"
+}
+
+interface RemoveAmountAction {
+  type: "REMOVE_AMOUNT"
+  payload: {
+    id: string
+    amount: number
+  }
+}
+
+export type CartReducerAction = AddItemAction | AddItemAmountAction | RemoveItemAction | ClearCartAction | RemoveAmountAction;
 
 // reducer
-const cartReducer = (statePart = [], action) => {
+const cartReducer = (statePart: Store["cart"] = [], action: CartReducerAction) => {
+
   switch (action.type) {
-    case ADD_TO_CART:
+    case "ADD_TO_CART":
       return [...statePart, { ...action.payload }];
-    case ADD_AMOUNT:
-      return statePart.map((item) =>
-        item.id == action.payload.id
-          ? { ...item, amount: item.amount + action.payload.amount }
+    case "ADD_AMOUNT":
+      return statePart.map((item) => {
+        item.id === action.id
+          ? { ...item, amount: item.amount + action.amount }
           : { ...item }
-      );
-    case REMOVE_AMOUNT:
+      });
```

Comment:
Since you've added curly braces, the function no longer returns the expression, but only `undefined`. 

Also, I think the defensive shallow copying is excessive. I think we can safely replace all of the `{ ...item }` spreads. It is generally a good practice not to mutate state variables willy-nilly, so I suppose spreading for the sake of cloning in most cases is unnecessary